src/spec.c

```diff
@@ -2106,6 +2113,36 @@ static void Indexes_ScanProc(RedisModuleCtx *ctx, RedisModuleString *keyname, Re
   if (scanner->cancelled) {
     return;
   }
+
+  const char* error;
+  // Get the memory limit and memory usage
+  setMemoryInfo(ctx);
+  if(used_memory > 0.8 * memoryLimit) {
```

Comment:
Shouldn't we add some search config for that? (default could be 0.8)

Reply:
seems legit
@MeirShpilraien thoughts?

Reply:
Why not canceling it at 100%?